server/pairing/client.go

```diff
@@ -69,6 +76,12 @@ func NewBaseClient(c *ConnectionParams) (*BaseClient, error) {
 
 	if serverCert == nil {
 		certErrs = fmt.Errorf("failed to connect to any of given addresses. %w", certErrs)
+		if retryCount < maxRetries {
+			time.Sleep(2 * time.Second)
+			logger.Info("failed to connect to any of given addresses. Retrying...", zap.Error(certErrs))
+			retryCount++
+			goto loop
+		}
 		signal.SendLocalPairingEvent(Event{Type: EventConnectionError, Error: certErrs.Error(), Action: ActionConnect})
 		return nil, certErrs
 	}
```

Comment:
I wonder if we need to break some of this code into a new function:

```go
func findServerCert(c *ConnectionParams) (*x509.Certificate, error) {
	netIps, err := server.FindReachableAddressesForPairingClient(c.netIPs)
	if err != nil {
		return nil, err
	}

	for i := range netIps {
		u, err := c.URL(i)
		if err != nil {
			return nil, err
		}
		serverCert, err = getServerCert(u)
		if err != nil {
			var certErr string
			if certErrs != nil {
				certErr = certErrs.Error()
			}
			certErrs = fmt.Errorf("%sconnecting to '%s' failed: %s; ", certErr, u, err.Error())
			continue
		}
		baseAddress = u
		break
	}

	if serverCert == nil {
		certErrs = fmt.Errorf("failed to connect to any of given addresses. %w", certErrs)
		signal.SendLocalPairingEvent(Event{Type: EventConnectionError, Error: certErrs.Error(), Action: ActionConnect})
		return nil, certErrs
	}
	
	return serverCert, nil
}
```

Usage, would be something like this.
```
maxRetries := 3

for i := 0; i< maxRetries; i++ {
	serverCert, err = findServerCert(c)
	if err == nil {
		break
	}
}
```

What do you think @qfrank

Reply:
I may accept both since the loop body is short enough, but i can change it for you 🙂 @Samyoul